src/Jobs/AppendPaginatedToSheet.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Maatwebsite\Excel\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Builder as EloquentBuilder;
+use Illuminate\Database\Eloquent\Relations\Relation;
+use Illuminate\Database\Query\Builder;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Laravel\Scout\Builder as ScoutBuilder;
+use Maatwebsite\Excel\Concerns\FromQuery;
+use Maatwebsite\Excel\Files\TemporaryFile;
+use Maatwebsite\Excel\Jobs\Middleware\LocalizeJob;
+use Maatwebsite\Excel\Writer;
+
+class AppendPaginatedToSheet implements ShouldQueue
+{
+    use Queueable, Dispatchable, ProxyFailures, InteractsWithQueue;
+
+    /**
+     * @var TemporaryFile
+     */
+    public $temporaryFile;
+
+    /**
+     * @var string
+     */
+    public $writerType;
+
+    /**
+     * @var int
+     */
+    public $sheetIndex;
+
+    /**
+     * @var FromQuery
+     */
+    public $sheetExport;
+
+    /**
+     * @var int
+     */
+    public $page;
+
+    /**
+     * @var int
+     */
+    public $perPage;
+
+    /**
+     * @param  FromQuery  $sheetExport
+     * @param  TemporaryFile  $temporaryFile
+     * @param  string  $writerType
+     * @param  int  $sheetIndex
+     * @param  int  $page
+     * @param  int  $perPage
+     */
+    public function __construct(
+        FromQuery $sheetExport,
+        TemporaryFile $temporaryFile,
+        string $writerType,
+        int $sheetIndex,
+        int $page,
+        int $perPage
+    ) {
+        $this->sheetExport   = $sheetExport;
+        $this->temporaryFile = $temporaryFile;
+        $this->writerType    = $writerType;
+        $this->sheetIndex    = $sheetIndex;
+        $this->page          = $page;
+        $this->perPage       = $perPage;
+    }
+
+    /**
+     * Get the middleware the job should be dispatched through.
+     *
+     * @return array
+     */
+    public function middleware()
+    {
+        return (method_exists($this->sheetExport, 'middleware')) ? $this->sheetExport->middleware() : [];
+    }
+
+    /**
+     * @param  Writer  $writer
+     *
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     * @throws \PhpOffice\PhpSpreadsheet\Reader\Exception
+     */
+    public function handle(Writer $writer)
+    {
+        (new LocalizeJob($this->sheetExport))->handle($this, function () use ($writer) {
+            $writer = $writer->reopen($this->temporaryFile, $this->writerType);
+
+            $sheet = $writer->getSheetByIndex($this->sheetIndex);
+
+            $sheet->appendRows($this->chunk($this->sheetExport->query()), $this->sheetExport);
+
+            $writer->write($this->sheetExport, $this->temporaryFile, $this->writerType);
+        });
+    }
+
+    /**
+     * @param  Builder|Relation|EloquentBuilder|ScoutBuilder  $query
+     */
+    protected function chunk($query)
+    {
+        if ($query instanceof \Laravel\Scout\Builder) {
```

Comment:
@Cellard  this breaks for everyone not using Laravel scout but do use laravel Excel :(

Reply:
Didn't break anything for me. I'm not using Laravel Scout, everything still works like expected